common/sgx/collateral.c

```diff
@@ -194,6 +195,46 @@ oe_result_t oe_get_sgx_quote_verification_collateral_from_certs(
     return result;
 }
 
+static unsigned char _hex_digit_to_num(char c)
+{
+    if (isdigit(c))
+        return (unsigned char)(c - '0');
+
+    if (c >= 'A' && c <= 'F')
+        return (unsigned char)(10 + (c - 'A'));
+
+    return (unsigned char)(10 + (c - 'a'));
+}
+
+static oe_result_t _hex_to_raw(
+    const char* hex_string,
+    size_t hex_string_size,
+    uint8_t* raw_buffer,
+    size_t raw_buffer_size)
+{
+    if (raw_buffer_size < hex_string_size / 2 + (hex_string_size % 2))
+    {
+        return OE_BUFFER_TOO_SMALL;
+    }
+
+    unsigned char v;
+
+    for (size_t i = 0; i < hex_string_size - 1; i += 2)
+    {
+        v = (unsigned char)(16 * _hex_digit_to_num(hex_string[i]) + _hex_digit_to_num(hex_string[i + 1]));
```

Comment:
Consider making it clear that this is a construction of single byte

```suggestion
        v = (unsigned char)(_hex_digit_to_num(hex_string[i]) << 4 | _hex_digit_to_num(hex_string[i + 1]));
```